Libraries/Components/WKWebViewIOS/WKWebViewIOS.ios.js

```diff
@@ -0,0 +1,485 @@
+        /*
+         * The HTTP body to send with the request. This must be a valid
+         * UTF-8 string, and will be sent exactly as specified, with no
+         * additional encoding (e.g. URL-escaping or base64) applied.
+         * NOTE: On Android, this can only be used with POST requests.
+         */
+        body: PropTypes.string,
+      }),
+      PropTypes.shape({
+        /*
+         * A static HTML page to display in the WebView.
+         */
+        html: PropTypes.string,
+        /*
+         * The base URL to be used for any relative links in the HTML.
+         */
+        baseUrl: PropTypes.string,
+      }),
+      /*
+       * Used internally by packager.
+       */
+      PropTypes.number,
+    ]),
+
+    /**
+     * Function that returns a view to show if there's an error.
+     */
+    renderError: PropTypes.func, // view to show if there's an error
+
+    /**
+     * Function that returns a loading indicator.
+     */
+    renderLoading: PropTypes.func,
+
+    /**
+     * Function that is invoked when the `WebView` has finished loading.
+     */
+    onLoad: PropTypes.func,
+
+    /**
+     * Function that is invoked when the `WebView` load succeeds or fails.
+     */
+    onLoadEnd: PropTypes.func,
+
+    /**
+     * Function that is invoked when the `WebView` starts loading.
+     */
+    onLoadStart: PropTypes.func,
+
+    /**
+     * Function that is invoked when the `WebView` load fails.
+     */
+    onError: PropTypes.func,
+
+    /**
+     * Boolean value that determines whether the web view bounces
+     * when it reaches the edge of the content. The default value is `true`.
+     * @platform ios
+     */
+    bounces: PropTypes.bool,
+
+    /**
+     * Boolean value that determines whether scrolling is enabled in the
+     * `WebView`. The default value is `true`.
+     * @platform ios
+     */
+    scrollEnabled: PropTypes.bool,
+
+    /**
+     * Controls whether to adjust the content inset for web views that are
+     * placed behind a navigation bar, tab bar, or toolbar. The default value
+     * is `true`.
+     */
+    automaticallyAdjustContentInsets: PropTypes.bool,
+
+    /**
+     * The amount by which the web view content is inset from the edges of
+     * the scroll view. Defaults to {top: 0, left: 0, bottom: 0, right: 0}.
+     */
+    contentInset: EdgeInsetsPropType,
+
+    /**
+     * Function that is invoked when the `WebView` loading starts or ends.
+     */
+    onNavigationStateChange: PropTypes.func,
+
+    /**
+     * A function that is invoked when the webview calls `window.postMessage`.
+     * Setting this property will inject a `postMessage` global into your
+     * webview, but will still call pre-existing values of `postMessage`.
+     *
+     * `window.postMessage` accepts one argument, `data`, which will be
+     * available on the event object, `event.nativeEvent.data`. `data`
+     * must be a string.
+     */
+    onMessage: PropTypes.func,
```

Comment:
Hmm yeah that's a good point, it would be kinda annoying. Another thing to consider is that it looks like `window.postMessage` is already set in WKWebView but has different behaviour to what we expected in WebView (https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage).

So we'd be guaranteed to be overwriting a native function although it *probably* wouldn't matter because the injected code will call back to it too.

I'll take a look at doing this.